libs/cart/testing/src/in-memory-backend/cart-root.service.integration.spec.ts

```diff
@@ -185,4 +199,71 @@ describe('DaffInMemoryBackendCartRootService | Integration', () => {
       expect(result.id).toBeDefined();
     });
   });
+
+  // cart coupon
+  describe('processing a list coupons request', () => {
+    let result;
+
+    beforeEach(done => {
+      httpClient.get<any>(`/api/cart-coupon/${cartId}/`).subscribe(res => {
+        result = res
+        done();
+      });
+    });
+
+    it('should return the cart coupons', () => {
+      expect(result).toEqual([mockCartCoupon]);
+    });
+  });
+
+  describe('processing an apply coupon request', () => {
+    let result;
+
+    beforeEach(done => {
+      mockCart.coupons = [];
+      httpClient.post<any>('commands/resetDb', {carts: [mockCart]}).subscribe(() => done());
+
+      httpClient.post<any>(`/api/cart-coupon/${cartId}/`, mockCartCoupon).subscribe(res => {
+        result = res
+        done();
+      });
+    });
+
+    it('should return a cart with the added item', () => {
```

Comment:
This test isn't quite right, I think. The same coupon is added to the returned cart on line 51, so posting with another add for the same coupon won't verify that the code works. You'd need to do it with a different coupon